Content.Server/Explosion/TriggerSystem.cs

```diff
@@ -30,17 +33,26 @@ public TriggerEvent(IEntity triggered, IEntity? user = null)
     [UsedImplicitly]
     public sealed class TriggerSystem : EntitySystem
     {
+        [Dependency] private readonly IGameTiming _gameTiming = default!;
+
         public override void Initialize()
         {
             base.Initialize();
             SubscribeLocalEvent<TriggerOnCollideComponent, StartCollideEvent>(HandleCollide);
+            SubscribeLocalEvent<TriggerOnProximityComponent, StartCollideEvent>(HandleCollide);
 
             SubscribeLocalEvent<DeleteOnTriggerComponent, TriggerEvent>(HandleDeleteTrigger);
             SubscribeLocalEvent<SoundOnTriggerComponent, TriggerEvent>(HandleSoundTrigger);
             SubscribeLocalEvent<ExplodeOnTriggerComponent, TriggerEvent>(HandleExplodeTrigger);
             SubscribeLocalEvent<FlashOnTriggerComponent, TriggerEvent>(HandleFlashTrigger);
 
             SubscribeLocalEvent<ExplosiveComponent, DestructionEventArgs>(HandleDestruction);
+            SubscribeLocalEvent<TriggerOnProximityComponent, ComponentInit>(CheckEnable);
+        }
+
+        private void CheckEnable(EntityUid uid, TriggerOnProximityComponent component, ComponentInit args)
+        {
+            component.Enabled = component.enabled;
```

Comment:
this is a weird way to invoke SetProximityFixture, just invoke it directly here

Reply:
That's for serialization , so if you set enabled to false on the yaml, it won't activate when first placed, sloth recommended it this way IIRC.